File: chrome-extension/generate-icons.js
// Simple script to generate placeholder icons for Chrome extension
const fs = require('fs');
const path = require('path');

// Minimal valid PNG file (1x1 transparent pixel)
// This is a base64-encoded minimal PNG
const minimalPNG = Buffer.from(
  'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==',
  'base64'
);

// Create a simple colored PNG using a minimal approach
// We'll create a simple gradient icon programmatically
function createIcon(size) {
  // For now, we'll create a simple colored square
  // In a real scenario, you'd use a library like 'sharp' or 'canvas'
  // But for quick setup, we'll create a minimal valid PNG
  
  // Create a simple SVG and convert to PNG would be ideal, but for now
  // let's create a script that generates icons using Node.js canvas if available
  // or provide instructions
  
  // For immediate fix, we'll create a minimal PNG placeholder
  return minimalPNG;
}

// Generate all required icon sizes
const sizes = [16, 32, 48, 128];
const iconsDir = path.join(__dirname, 'icons');

// Ensure icons directory exists
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir, { recursive: true });
}

console.log('Generating placeholder icons...');

// For a proper solution, we need to create actual PNG files
// Let's create a better solution using a simple approach
// We'll write a script that can be run with node-canvas or use an online tool

// Create a simple HTML file that generates proper icons
const iconGeneratorHTML = `<!DOCTYPE html>
<html>
<head>
  <title>Generate Icons</title>
  <style>
    body { font-family: Arial; padding: 20px; background: #0a0a0f; color: #fafafa; }
    canvas { border: 1px solid #333; margin: 10px; background: #1a1a2e; }
    button { padding: 10px 20px; background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); 
             color: white; border: none; border-radius: 8px; cursor: pointer; margin: 5px; }
  </style>
</head>
<body>
  <h1>VerseForge Icon Generator</h1>
  <p>Right-click each icon below and "Save image as..." to save to icons/ folder</p>
  ${sizes.map(size => `<canvas id="icon${size}" width="${size}" height="${size}"></canvas>`).join('')}
  <br><button onclick="generate()">Generate Icons</button>
  <script>
    function generate() {
      const sizes = [${sizes.join(',')}];
      sizes.forEach(size => {
        const canvas = document.getElementById('icon' + size);
        const ctx = canvas.getContext('2d');
        const gradient = ctx.createLinearGradient(0, 0, size, size);
        gradient.addColorStop(0, '#667eea');
        gradient.addColorStop(1, '#764ba2');
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, size, size);
        ctx.strokeStyle = '#ffffff';
        ctx.lineWidth = Math.max(1, size / 16);
        const center = size / 2;
        ctx.beginPath();
        ctx.moveTo(center - size * 0.3, center - size * 0.3);
        ctx.lineTo(center + size * 0.3, center + size * 0.3);
        ctx.moveTo(center + size * 0.3, center - size * 0.3);
        ctx.lineTo(center - size * 0.3, center + size * 0.3);
        ctx.stroke();
        ctx.beginPath();
        ctx.arc(center, center, size * 0.1, 0, Math.PI * 2);
        ctx.fillStyle = '#ffffff';
        ctx.fill();
      });
    }
    window.onload = generate;
  </script>
</body>
</html>`;

fs.writeFileSync(path.join(__dirname, 'generate-icons.html'), iconGeneratorHTML);
console.log('✅ Created generate-icons.html');
console.log('');
console.log('📝 To generate icons:');
console.log('1. Open generate-icons.html in your browser');
console.log('2. Right-click each icon and "Save image as..."');
console.log('3. Save them as: icon16.png, icon32.png, icon48.png, icon128.png');
console.log('4. Place them in the icons/ folder');
console.log('');
console.log('⚠️  For now, creating minimal placeholder icons...');

// Create minimal placeholder PNGs (transparent 1x1, but Chrome needs actual files)
// We'll create a simple colored square using a different approach
// Actually, let's use a Python script or create proper PNGs

// For immediate fix, let's create a script that uses ImageMagick or similar
// But the simplest is to use the HTML generator

console.log('✅ Use generate-icons.html to create proper icons!');

File: chrome-extension/generate-icons.py
#!/usr/bin/env python3
"""
Generate Aether Chrome Extension Icons
Resizes logo.png to required sizes
"""

from PIL import Image
import os
import sys

def main():
    sizes = [16, 32, 48, 128]
    icons_dir = 'icons'
    source_image = 'logo.png'
    
    if not os.path.exists(source_image):
        print(f"❌ Error: {source_image} not found.")
        sys.exit(1)
        
    # Create icons directory if it doesn't exist
    os.makedirs(icons_dir, exist_ok=True)
    
    print("Generating Aether extension icons from logo.png...")
    
    try:
        img = Image.open(source_image)
        # Convert to RGBA if not already
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
            
        for size in sizes:
            # Resize with high quality resampling
            # Use LANCZOS for better compatibility across PIL versions
            try:
                # Try new Resampling enum first (Pillow >= 10.0.0)
                resample_method = Image.Resampling.LANCZOS
            except AttributeError:
                # Fall back to old constant (Pillow < 10.0.0)
                resample_method = Image.LANCZOS
            
            icon = img.resize((size, size), resample_method)
            filename = f'{icons_dir}/icon{size}.png'
            icon.save(filename, 'PNG')
            print(f"✅ Created {filename}")
            
        print("\n🎉 All icons generated successfully!")
        
    except Exception as e:
        print(f"❌ Error processing image: {e}")
        sys.exit(1)

if __name__ == '__main__':
    main()
